### PR TITLE
Implicit labels

### LINKED ISSUE #2: add implicit labels to prod strategy
E.g.:

```json
"Labels": {
    "Architecture": "x86_64",
    "Build_Host": "build.host.redhat.com",
    "Name": "rhel-server-docker",
    "Release": "4",
    "Vendor": "Red Hat, Inc.",
    "Version": "7.1"
}
```
This should be configurable.

### PR DESCRIPTION
**Pull request by twaugh**
This adds implicit Dockerfile labels for `Architecture`, `Build_Host`, `Vendor`, and `Authoritative_Registry`. Fixes #2.

### CODE REVIEW

**Comment by TomasTomecek** [comment]
Went through the code more thoroughly. Looks just fine to me. I think it's ready to merge. Thanks for doing this Tim!

I'm planning to merge it tomorrow. @mmilata is planning to issue a new release regarding issues with authentication. Once that's done, I'll merge it (want to be super sure we don't break anything -- this is a significant feature and would like to test it before putting it to prod). Am also planning to write some unit tests since we have CI up and running. The main reason for me writing the tests is to get familiar with the new code base in this patch (I think that `class DockJsonManipulator` would deserve precise tests)

File: osbs/build.py [hunk 1]
```diff
@@ -31,16 +31,31 @@ def get_dock_json(self):
...
+        return match[0]
+
+    def dock_json_set_arg(self, plugin_type, plugin_name, arg_key, arg_value):
+        plugin_conf = self._dock_json_get_plugin_conf (plugin_type, plugin_name, arg_key)
         plugin_conf['args'][arg_key] = arg_value
 
+    def dock_json_merge_arg(self, plugin_type, plugin_name, arg_key, arg_dict):
+        plugin_conf = self._dock_json_get_plugin_conf (plugin_type, plugin_name, arg_key)
+
+        # Values from the template JSON override our implicit values
```
Comment (TomasTomecek) [inline osbs/build.py]: I think that I would be in favor of doing this the other way around: `arg_value` overrides template -- if someone wants to override a value, we shouldn't stop him/her. What do you think Tim?

File: osbs/build.py [hunk 2]
```diff
@@ -172,6 +187,10 @@ def __init__(self, **kwargs):
             BuildParam('kojiroot', True),
             BuildParam('kojihub', True),
             BuildParam('sources_command', True),
+            BuildParam('architecture', True),
```
Comment (TomasTomecek) [inline osbs/build.py]: could you please add these new arguments to docstring above?

File: osbs/build.py (hunk 1 above)
Comment (twaugh) [inline osbs/build.py]: Yes, makes sense. I've committed that change on `next`.

**Comment by TomasTomecek** [comment]
Merged via `next` branch. Thanks a lot Tim, great job!

**Pull request closed without merging**